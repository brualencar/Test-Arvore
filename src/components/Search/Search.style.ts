import { styled } from 'styled-components';

export const Wrapper = styled.div``;

export const Input = styled.input`
  width: 100%;
  height: 35px;
  border-radius: 20px;
  border: 1px solid ${({ theme }) => theme.colors.background.neutral.default};
  background: ${({ theme }) => theme.colors.background.neutral.light};

  &:focus {
    outline: none;
    padding-left: ${({ theme }) => theme.spacing.km4};
  }

  &:focus::placeholder {
    color: transparent;
  }

  &::placeholder {
    color: ${({ theme }) => theme.colors.text.neutral.default};
    font-size: ${({ theme }) => theme.font.size.base};
    font-weight: ${({ theme }) => theme.font.weight.regular};
    padding-left: ${({ theme }) => theme.spacing.km4};
  }
`;
