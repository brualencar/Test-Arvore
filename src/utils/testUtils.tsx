import React, { PropsWithChildren } from 'react';

import { ThemeProvider } from 'styled-components';

import { render as rtlRender, RenderOptions } from '@testing-library/react';

import { theme } from '../assets/styles/globalStyle';

function render(ui: React.ReactElement, renderOptions?: RenderOptions) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
