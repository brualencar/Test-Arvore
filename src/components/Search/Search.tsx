import React, { useState } from 'react';

import * as Styled from './Search.style';
import { InputEventProps } from './Search.types';

function Search() {
  const [search, setSearch] = useState('');

  const eventOnInput = (event: React.ChangeEvent<InputEventProps>) => {
    const target = event.target as HTMLInputElement;
    setSearch(target.value);
  };

  const handleKeyPress = (event: React.KeyboardEvent) => {
    if (event.key === 'Enter') {
      window.location.assign(`/search-result?search=${search}`);
    }
  };

  return (
    <Styled.Wrapper data-testid='search'>
      <Styled.Input
        data-testid='search-input'
        type='search'
        placeholder='Search'
        onChange={eventOnInput}
        onKeyDown={handleKeyPress}
      />
    </Styled.Wrapper>
  );
}

export default Search;
