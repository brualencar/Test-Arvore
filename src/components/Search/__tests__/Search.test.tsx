import React from 'react';

import { fireEvent, render, screen } from '../../../utils/testUtils';
import Search from '../Search';

describe('Search', () => {
  it('Should match snapshot Search component', () => {
    const { container } = render(<Search />);

    expect(container).toMatchSnapshot();
  });
  it('Should render Search component', () => {
    render(<Search />);

    const divElement = screen.getByTestId('search');
    expect(divElement).toBeInTheDocument();
  });

  it('calls Search with the correct search value', async () => {
    render(<Search />);

    const input = await screen.getByTestId('search-input');
    fireEvent.change(input, { target: { value: 'Book' } });

    expect(input).toHaveValue('Book');
  });

  it('calls Search and redirect to search result page', async () => {
    const handleKeyPress = jest.fn();
    Object.defineProperty(window, 'location', {
      value: { assign: handleKeyPress },
    });
    render(<Search />);

    const input = await screen.getByTestId('search-input');
    fireEvent.change(input, { target: { value: 'Book' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 13, charCode: 13 });

    expect(handleKeyPress).toBeCalledWith(`/search-result?search=Book`);
  });
});
