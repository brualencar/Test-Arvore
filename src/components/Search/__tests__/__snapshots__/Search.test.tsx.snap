// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should match snapshot Search component 1`] = `
<div>
  <div
    class="sc-bdDrbm"
    data-testid="search"
  >
    <input
      class="sc-gtcAbF ZeuDL"
      data-testid="search-input"
      placeholder="Search"
      type="search"
    />
  </div>
</div>
`;
